packages/core/src/transactions/TransactionIntent.ts

```diff
@@ -0,0 +1,45 @@
+import TransactionPath from './TransactionPath'
+import TransactionRequest from './TransactionRequest'
+import { calculateTransactionPath } from './utils/simplePath'
+import Organization from '../wrappers/Organization'
+
+export interface TransactionIntentData {
+  contractAddress: string
+  functionName: string
+  functionArgs: string[]
+}
+
+export default class TransactionIntent {
+  readonly contractAddress!: string
+  readonly functionName!: string
+  readonly functionArgs!: string[]
+  #org: Organization
+
+  constructor(data: TransactionIntentData, org: Organization) {
+    this.#org = org
+
+    Object.assign(this, data)
+  }
+
+  async paths(
+    address: string,
+    { as, path }: { as?: string; path?: string[] }
+  ): Promise<TransactionPath[]> {
+    return calculateTransactionPath(
+      address,
+      this.contractAddress,
+      this.functionName,
+      this.functionArgs,
+      this.#org,
+      path,
+      as
+    )
```

Comment:
I’m wondering what we should do when `calculateTransactionPath` returns `null` here… should we throw?

Reply:
I choose to return null when there is no path found. So the logic that calls the `calculateTransactionPath` if receive `null` doesn't even create a new instance of `TransactionPath`. What do you think?

Reply:
Should we return an empty array?